druid/src/widget/slider.rs

```diff
@@ -68,7 +78,18 @@ impl Slider {
         let scalar = ((mouse_x + self.x_offset - knob_width / 2.) / (slider_width - knob_width))
             .max(0.0)
             .min(1.0);
-        self.min + scalar * (self.max - self.min)
+        let mut value = self.min + scalar * (self.max - self.min);
+        if let Some(stepping) = self.step {
+            let steps = if stepping > 0.0 {
```

Comment:
If you move this check into the `with_step` function this could be simplified a lot. If stepping is `0.0` it is probably a bug, and a warning should be logged or something like that.